### DISCUSSION TITLE
How to import NuxtLink for use in dynanic component (:is="" attribute)

### DISCUSSION THREAD

**Discussion by farin** [nuxt/nuxt#16992] [Questions] [answered]
I am using component which is conditionally clickable. So template looks like

`<component :is="link ? NuxtLink : 'div'">...`

In this case, NuxtLink is not auto imported and must be imported in script setup
with rc8 following works
`import NuxtLink from '~~/node_modules/nuxt/dist/app/components/nuxt-link'`
replaced my older code (which is currently not working)
`import { NuxtLink } from '~~/.nuxt/components'`

Is there proper (and stable) way how to do it?

**Comment by tylerforesthauser** [comment]
I believe you just have to use 'NuxtLink' as a string not a variable. There should be no need to import it.

**Reply by farin** [reply]
no, that isn't working. it just create unknown <nuxtlink> element which is not bound to real NuxtLink component

**Comment by danielroe** [comment, marked as answer]
You can use it like this:

```js
const NuxtLink = resolveComponent('NuxtLink')
```

https://v3.nuxtjs.org/guide/directory-structure/components/#dynamic-components

**Reply by MorevM** [reply]
@danielroe is there an option to make it available without `resolveComponent` using string like `:is="cond ? 'nuxt-link' : 'div'"`? 
Was unexpected for me (and i guess others), because it's worked fine with Nuxt 2 and Nuxt own components are declared as global.

**Reply by danielroe** [reply]
No, Nuxt's own components aren't declared as global. They are auto imported directly when they are used, which is the most performant option.

If you need a link component to be globally available (ie you can use a string to reference it), you could create your own global wrapper for it, e.g. `~/components/MyLink.global.vue`.
